src/index.ts

```diff
@@ -27,15 +30,27 @@ const createGoError = <E extends Error>(err: unknown): GoResultError<E> => {
   return fail(new Error('' + err));
 };
 
-export const go = async <T, E extends Error>(fn: Promise<T> | (() => Promise<T>)): Promise<GoResult<T, E>> => {
-  // We need try/catch because `fn` might throw sync errors as well
+export const go = async <T, E extends Error>(
+  fn: () => Promise<T>,
```

Comment:
> Oh, I didn't expect that you'll start implementing the tuple version of the `go` function sweat_smile We've decided that we'll switch to returning [an object](https://github.com/api3dao/promise-utils/pull/2#discussion_r812124852). Sorry, I could have mentioned that confused Can you remove that part? I'll review the rest (the retry functionality) once that's gone.

No problem, I had a feeling this might be the case :sweat_smile:

Fixed now and I think the only change major change would be to `fn` here.